cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -334,6 +334,12 @@ public void destroy() {
         alertPublisher.publish(trackedIterator);
       } catch (Exception e) {
         LOG.warn("Stage {} had errors publishing alerts. Alerts may not have been published.", name, e);
+      } finally {
+        try {
+          alertPublisher.destroy();
```

Comment:
we call it already in spark. we should remove that then, right? Or do we not use SmartWorkflow in that case?